Python/Product/PythonTools/PythonTools/Intellisense/CompletionAnalysis.cs

```diff
@@ -92,18 +93,20 @@ internal class CompletionAnalysis {
                 _options.MemberOptions,
                 triggerChar == '\0' ? Analysis.LanguageServer.CompletionTriggerKind.Invoked : Analysis.LanguageServer.CompletionTriggerKind.TriggerCharacter,
                 triggerChar == '\0' ? null : triggerChar.ToString()
-            ), "GetCompletions.GetMembers");
+            ), "GetCompletions.GetMembers", null, pyReplEval?.Analyzer == null ? 1 : 5);
 
-            if (completions?.items == null) {
+            if (completions == null) {
                 return null;
             }
 
             var snapshotSpan = completions._applicableSpan.HasValue
                 ? ((SourceSpan)completions._applicableSpan.Value).ToSnapshotSpan(_snapshot)
                 : new SnapshotSpan(point, 0);
             _span = bufferInfo.CurrentSnapshot.CreateTrackingSpan(snapshotSpan, SpanTrackingMode.EdgeInclusive);
-
-            var members = completions.items.Select(c => new CompletionResult(
+            var members = completions.items.MaybeEnumerate().Select(c => new CompletionResult(
+                // if a method stub generation (best guess by comparing length),
+                // merge entry based on label, for everything else, use insertion text
+                c.filterText ?? ((c.insertText ?? c.label).Length > c.label.Length ? c.label : c.insertText ?? c.label),
```

Comment:
`c.filterText ?? (c.insertText != null && c.insertText.Length <= c.label.Length ? c.insertText : c.label)` - a bit longer, but much easier to understand.